src/contracts/pack.ts

```diff
@@ -121,448 +126,428 @@ export class Pack implements UpdateableNetwork {
+      erc20Rewards,
+      erc721Rewards,
+      erc1155Rewards,
     };
   }
 
-  public async transferBatchFrom(
-    from: string,
-    to: string,
-    args: IPackBatchArgs[],
-    data: BytesLike = [0],
-  ): Promise<TransactionResult> {
-    const ids = args.map((a) => a.tokenId);
-    const amounts = args.map((a) => a.amount);
-    return {
-      receipt: await this.contractWrapper.sendTransaction(
-        "safeBatchTransferFrom",
-        [from, to, ids, amounts, data],
-      ),
-    };
-  }
+  /** *****************************
+   * PRIVATE FUNCTIONS
+   *******************************/
 
-  public async setApproval(
-    operator: string,
-    approved = true,
-  ): Promise<TransactionResult> {
-    return {
-      receipt: await this.contractWrapper.sendTransaction("setApprovalForAll", [
-        operator,
-        approved,
-      ]),
-    };
-  }
+  private async toPackContentArgs(metadataWithRewards: PackMetadataOutput) {
+    const contents: ITokenBundle.TokenStruct[] = [];
+    const numOfRewardUnits = [];
+    const { erc20Rewards, erc721Rewards, erc1155Rewards } = metadataWithRewards;
 
-  public async depositLink(amount: BigNumberish): Promise<TransactionResult> {
-    const chainId = await this.contractWrapper.getChainID();
-    const chainlink = ChainlinkVrf[chainId];
-    const erc20 = new Contract(
-      chainlink.linkTokenAddress,
-      ERC20Abi,
-      this.contractWrapper.getProvider(),
-    ) as IERC20;
-    // TODO: make it gasless
-    const tx = await erc20.transfer(
-      this.getAddress(),
-      amount,
-      await this.contractWrapper.getCallOverrides(),
-    );
-    return { receipt: await tx.wait() };
-  }
+    const provider = this.contractWrapper.getProvider();
+    const owner = await this.contractWrapper.getSignerAddress();
 
-  // TODO new withdraw LINK function in contract
-  // public async withdrawLink(to: string, amount: BigNumberish) {
-  //   const chainId = await this.contractWrapper.getChainID();
-  //   const chainlink = ChainlinkVrf[chainId];
-  //   await this.contractWrapper.sendTransaction("transferERC20", [
-  //     chainlink.linkTokenAddress,
-  //     to,
-  //     amount,
-  //   ]);
-  // }
+    for (const erc20 of erc20Rewards) {
+      const normalizedQuantity = await normalizePriceValue(
+        provider,
+        erc20.quantityPerReward,
+        erc20.contractAddress,
+      );
+      // Multiply the quantity of one reward by the number of rewards
+      const totalQuantity = normalizedQuantity.mul(erc20.totalRewards);
+      const hasAllowance = await hasERC20Allowance(
+        this.contractWrapper,
+        erc20.contractAddress,
+        totalQuantity,
+      );
+      if (!hasAllowance) {
+        throw new Error(
+          `ERC20 token with contract address "${
+            erc20.contractAddress
+          }" does not have enough allowance to transfer.\n\nYou can set allowance to the multiwrap contract to transfer these tokens by running:\n\nawait sdk.getToken("${
```

Comment:
There are some key differences because of the pack interface we want vs the one the contract implements, we need to do some multiplication/division in all these in the pack file that's different for each function in pack + different from multiwrap - which is why I opted to keep these all separate as verbose as it is, b/c handling all the casing seems messier and wanted to keep things explicit.

TL;DR, because contract uses total tokens + number of rewards (division), but we use tokens per reward + number of rewards (multiplication) we have to do some conversions between these interfaces which makes things different from MW and each other.

Reply:
i see - that makes sense.